--- vmmon-only/linux/vmmonInt.h.1	2015-11-10 20:57:44.904945120 +0900
+++ vmmon-only/linux/vmmonInt.h	2015-11-10 20:57:33.089099902 +0900
@@ -28,11 +28,11 @@
  * Hide all kernel compatibility stuff in these macros and functions.
  */
 
-#ifdef VMW_HAVE_SMP_CALL_3ARG
+//#ifdef VMW_HAVE_SMP_CALL_3ARG
 #define compat_smp_call_function(fn, info, wait) smp_call_function(fn, info, wait)
-#else
-#define compat_smp_call_function(fn, info, wait) smp_call_function(fn, info, 1, wait)
-#endif
+//#else
+//#define compat_smp_call_function(fn, info, wait) smp_call_function(fn, info, 1, wait)
+//#endif
 
 /*
  * Although this is not really related to kernel-compatibility, I put this
